notebooks/pytorch/training.ipynb

```diff
@@ -0,0 +1,306 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# PyTorch Training\n",
+    "This notebook trains a model to predict whether the given sonar signals are bouncing off a metal cylinder or off a cylindrical rock from [UCI Machine Learning Repository](https://archive.ics.uci.edu/ml/datasets/Connectionist+Bench+%28Sonar%2C+Mines+vs.+Rocks%29).\n",
+    "\n",
+    "## The data\n",
+    "The Sonar Signals dataset that this sample uses for training is provided by the UC Irvine Machine Learning Repository. Google has hosted the data on a public GCS bucket `gs://cloud-samples-data/ml-engine/sonar/sonar.all-data`.\n",
+    "\n",
+    "* `sonar.all-data` is split for both training and evaluation\n",
+    "\n",
+    "Note: Your typical development process with your own data would require you to upload your data to GCS so that you can access that data from inside your notebook. However, in this case, Google has put the data on GCS to avoid the steps of having you download the data from UC Irvine and then upload the data to GCS.\n",
+    "\n",
+    "### Disclaimer\n",
+    "This dataset is provided by a third party. Google provides no representation, warranty, or other guarantees about the validity or any other aspects of this dataset.\n",
+    "\n",
+    "# Build your model\n",
+    "First, you'll create the model (provided below). This is similar to your normal process for creating a PyTorch model. However, there is one key difference:\n",
+    "\n",
+    "Downloading the data from GCS at the start of your file, so that you can access the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from google.cloud import storage\n",
+    "import pandas as pd\n",
+    "import torch\n",
+    "import torch.optim as optim\n",
+    "import torch.nn as nn\n",
+    "from torch.utils.data import Dataset\n",
+    "from torch.utils.data import DataLoader\n",
+    "from torch.utils.data import random_split"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ---------------------------------------\n",
+    "# Add code to download the data from GCS (in this case, using the publicly hosted data).\n",
+    "# you will then be able to use the data when training your model.\n",
+    "# ---------------------------------------\n",
+    "# Public bucket holding the census data\n",
+    "bucket = storage.Client().bucket('cloud-samples-data')\n",
+    "\n",
+    "# Path to the data inside the public bucket\n",
+    "blob = bucket.blob('ml-engine/sonar/sonar.all-data')\n",
+    "# Download the data\n",
+    "blob.download_to_filename('sonar.all-data')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ---------------------------------------\n",
+    "# Read in the data\n",
```

Comment:
Nit: can we move the comments in Code Cell to a Markup Cell. Same below

Reply:
Done